File: matlab_and_octave/examples/all_regression.m
% Explicit examples on how to use regressions

size_cache=10;
width=2.1;
C=0.017;
tube_epsilon=1e-2;
len=42;
num=20;
dist=2.3;

trainlab=[ones(1,num*2) -ones(1,num*2)];
testlab=[ones(1,num*2) -ones(1,num*2)];

traindata=[randn(2,num)-dist, randn(2,num)+dist, randn(2,num)+dist*[ones(1,num); zeros(1,num)], randn(2,num)+dist*[zeros(1,num); ones(1,num)]];
testdata=[randn(2,num+7)-dist, randn(2,num+7)+dist, randn(2,num+7)+dist*[ones(1,num+7); zeros(1,num+7)], randn(2,num+7)+dist*[zeros(1,num+7); ones(1,num+7)]];


%
% svm-based
%

% SVR Light
disp('SVRLight');

sg('set_features', 'TRAIN', traindata);
sg('set_kernel', 'GAUSSIAN', 'REAL', size_cache, width);
sg('init_kernel', 'TRAIN');

sg('set_labels', 'TRAIN', trainlab);
sg('new_regression', 'SVRLIGHT');
sg('svr_tube_epsilon', tube_epsilon);
sg('c', C);
sg('train_regression');

sg('set_features', 'TEST', testdata);
sg('set_labels', 'TEST', testlab);
sg('init_kernel', 'TEST');
result=sg('classify');


% LibSVR
disp('LibSVR');

sg('set_features', 'TRAIN', traindata);
sg('set_kernel', 'GAUSSIAN', 'REAL', size_cache, width);
sg('init_kernel', 'TRAIN');

sg('set_labels', 'TRAIN', trainlab);
sg('new_regression', 'LIBSVR');
sg('svr_tube_epsilon', tube_epsilon);
sg('c', C);
sg('train_regression');

sg('set_features', 'TEST', testdata);
sg('set_labels', 'TEST', testlab);
sg('init_kernel', 'TEST');
result=sg('classify');


%
% misc
%

% KRR
disp('KRR');

sg('set_features', 'TRAIN', traindata);
sg('set_kernel', 'GAUSSIAN', 'REAL', size_cache, width);
sg('init_kernel', 'TRAIN');

sg('set_labels', 'TRAIN', trainlab);
sg('new_regression', 'KRR');
tau=1.2;
sg('krr_tau', tau);
sg('c', C);
sg('train_regression');

sg('set_features', 'TEST', testdata);
sg('init_kernel', 'TEST');
result=sg('classify');

File: examples/python-modular/all_regression.py
#!/usr/bin/env python
"""
Explicit examples on how to use regressions
"""

from numpy import array
from numpy.random import seed, rand
from shogun.Features import Labels, RealFeatures
from shogun.Kernel import GaussianKernel
from shogun.Regression import *

from tools.load import LoadMatrix
lm=LoadMatrix()
fm_train=lm.load_numbers('../data/fm_train_real.dat')
fm_test=lm.load_numbers('../data/fm_test_real.dat')
label_train=lm.load_labels('../data/label_train_oneclass.dat')

###########################################################################
# svm-based
###########################################################################

def svr_light ():
	print 'SVRLight'

	feats_train=RealFeatures(fm_train)
	feats_test=RealFeatures(fm_test)
	width=2.1
	kernel=GaussianKernel(feats_train, feats_train, width)

	C=0.017
	epsilon=1e-5
	tube_epsilon=1e-2
	num_threads=3
	labels=Labels(label_train)

	try:
		svr=SVRLight(C, epsilon, kernel, labels)
	except NameError:
		print 'No support for SVRLight available.'
		return

	svr.set_tube_epsilon(tube_epsilon)
	svr.parallel.set_num_threads(num_threads)
	svr.train()

	kernel.init(feats_train, feats_test)
	svr.classify().get_labels()

def libsvr ():
	print 'LibSVR'

	feats_train=RealFeatures(fm_train)
	feats_test=RealFeatures(fm_test)
	width=2.1
	kernel=GaussianKernel(feats_train, feats_train, width)

	C=0.017
	epsilon=1e-5
	tube_epsilon=1e-2
	num_threads=1
	labels=Labels(label_train)

	svr=LibSVR(C, epsilon, kernel, labels)
	svr.set_tube_epsilon(tube_epsilon)
	svr.parallel.set_num_threads(num_threads)
	svr.train()

	kernel.init(feats_train, feats_test)
	svr.classify().get_labels()

###########################################################################
# misc
###########################################################################

def krr ():
	print 'KRR'

	feats_train=RealFeatures(fm_train)
	feats_test=RealFeatures(fm_test)
	width=0.8
	kernel=GaussianKernel(feats_train, feats_train, width)

	C=0.42
	tau=1e-6
	num_threads=1
	labels=Labels(label_train)

	krr=KRR(tau, kernel, labels)
	krr.parallel.set_num_threads(num_threads)
	krr.train()

	kernel.init(feats_train, feats_test)
	krr.classify().get_labels()

###########################################################################
# call functions
###########################################################################

if __name__=='__main__':
	seed(42)

	svr_light()
	libsvr()

	krr()
